xonsh/environ.py

```diff
@@ -38,7 +34,11 @@ def wrapper(*args, **kwargs):
         except subprocess.CalledProcessError:
             return
         return func(*args, **kwargs)
-    return wrapper
+    if ON_WINDOWS:
```

Comment:
If we're on Windows, we don't even need to go to the trouble of defining the `wrapper` function if we're just going to throw it away.  I might recommend sliding this check up.  So rather than defining the function first and then doing the Windows check, we could do:
```
if ON_WINDOWS:
    return func

@wraps(func)
def wrapper
...
```